### ISSUE TITLE
Hot to prevent Turbolinks back button event?

### ISSUE THREAD

**Issue by lnikell** [turbolinks/turbolinks#27]
I'm using https://github.com/dimsemenov/PhotoSwipe like a photo gallery, but I can't using it with turbolinks because that library using html5 history too and when I using back button when gallery opened I do not want to trigger Turbolinks back event. 

I'm using Turbolinks 5.0.0.beta2

**Comment by packagethief** [comment]
Hi @lnikell. Turbolinks should only handle its own `popstate` events. It [adds its own state object when it pushes an entry](https://github.com/turbolinks/turbolinks/blob/master/src/turbolinks/history.coffee#L45), and [checks for that state on pop](https://github.com/turbolinks/turbolinks/blob/master/src/turbolinks/history.coffee#L33).

Can you provide any more details?

**Comment by lnikell** [comment]
Hi, @packagethief ! I'm using http://photoswipe.com/ and if you open the image you will that this library use pushstate and after your click on back button restore the last state. So when I use it with turbolinks and click on back button I receive rerender of the whole page. I mean that turbolinks trigger some events on back button click, but I do not want that he do it.

**Comment by 1989medium** [comment]
<%= link_to 'Back', :back %> Not Work...

**Comment by packagethief** [comment]
Hey @lnikell. I took photoswipe for a spin locally but I wasn't able to reproduce the issue you're describing.

It seems that photoswipe uses `pushState` when launching the gallery initially, and `replaceState` as you page through photos. Clicking "back" closes the gallery and restores the original page; clicking "forward" doesn't reopen it (though it does change the location hash). I observed the same behavior on the demo at http://photoswipe.com.

I tested by opening a gallery on click, and in response to a page load. At no time did clicking back result in a full page load. However, clicking back does result in a Turbolinks load. That's expected, since we pop back to a Turbolinks state, which performs a visit.

I did come across [this similar-sounding issue on photoswipe's issue tracker](https://github.com/dimsemenov/PhotoSwipe/issues/700), but again, I wasn't able to reproduce.
